evalml/pipelines/time_series_classification_pipelines.py

```diff
@@ -172,21 +187,22 @@ def threshold(self, value):
     def _predict(self, X, y, objective=None, pad=False):
         y_encoded = self._encode_targets(y)
         features = self.compute_estimator_features(X, y_encoded)
-        features_no_nan = features.dropna(axis=0, how="any")
+        features_no_nan, y_encoded = drop_rows_with_nans(features, y_encoded)
 
         if objective is not None:
             objective = get_objective(objective, return_instance=True)
             if not objective.is_defined_for_problem_type(self.problem_type):
                 raise ValueError(f"Objective {objective.name} is not defined for time series binary classification.")
 
         if self.threshold is None:
-            predictions = self.estimator.predict(features_no_nan)
+            predictions = self._estimator_predict(features_no_nan, y_encoded)
         else:
-            ypred_proba = self.estimator.predict_proba(features_no_nan).iloc[:, 1]
+            proba = self._estimator_predict_proba(features_no_nan, y_encoded)
+            proba = proba.iloc[:, 1]
```

Comment:
Why is this necessary? Not covered by `_estimator_predict_proba`?

Reply:
Yea, `_estimator_predict_proba` returns the probabilities for all classes. For binary pipelines in `predict`, we threshold the predictions for the "positive" class if the threshold is not None. Hence the need for `.iloc[:, 1]`.